src/main/kotlin/org/rust/lang/core/cfg/ControlFlowGraph.kt

```diff
@@ -134,34 +131,45 @@ sealed class ExitPoint {
     class TailStatement(val stmt: RsExprStmt) : ExitPoint()
 
     companion object {
-        fun process(fn: RsFunction, sink: (ExitPoint) -> Unit) {
-            fn.block?.acceptChildren(ExitPointVisitor(sink))
-        }
-
-        fun process(fn: RsLambdaExpr, sink: (ExitPoint) -> Unit) {
-            fn.expr?.acceptChildren(ExitPointVisitor(sink))
+        fun process(fn: PsiElement?, sink: (ExitPoint) -> Unit) {
+            fn?.acceptChildren(ExitPointVisitor(sink))
         }
     }
 }
 
 private class ExitPointVisitor(
     private val sink: (ExitPoint) -> Unit
 ) : RsVisitor() {
+    var inTry = false
```

Comment:
I think you need a counter instead because of nested `try`:
```Rust
fn foo() -> i32 {
    let a: Result<i32, ()> = try {
        let tmp: Result<i32, ()> = try { Err(())? };
        tmp? + Ok(42)? // wrong highlighting currently
    };
    return/*caret*/ 1;
}
```
Please add such tests